spec/features/super_admin/logging/view_logs_for_ip_spec.rb

```diff
@@ -0,0 +1,88 @@
+describe 'View authentication requests for an IP', type: :feature do
+  let(:ip) { '1.2.3.4' }
+  let(:username) { 'ABCDEF' }
+
+  let(:admin_user) { create(:user, organisation_id: organisation.id) }
+  let(:organisation) { create(:organisation) }
+  let(:location) { create(:location, organisation_id: organisation.id) }
+
+  let(:other_user) { create(:user, organisation_id: other_user_organisation.id) }
+  let(:other_user_organisation) { create(:organisation) }
+  let(:other_user_location) { create(:location, organisation_id: other_user_organisation.id) }
+
+  let(:super_admin) { create(:user, super_admin: true) }
+  let(:super_admin_organisation) { create(:organisation) }
+  let(:super_admin_location) { create(:location, super_admin_organisation: super_admin.organisation) }
+
+  before do
+    Session.create!(
+      start: 3.days.ago,
+      username: username,
+      siteIP: ip,
+      success: true
+    )
+
+    Session.create!(
+      start: 3.days.ago,
+      username: username,
+      siteIP: ip,
+      success: true
+    )
+
+    Session.create!(
+      start: 3.days.ago,
+      username: username,
+      siteIP: ip,
+      success: true
+    )
+
+    create(:ip, location_id: location.id, address: ip)
```

Comment:
This ip creation seems a bit convoluted to me. You are doing something like:
- setting a variable which has an ip address: `let(:ip) { '1.2.3.4' }`
- creating three sessions referencing that variable Session.create!(siteIP: ip)`
- then creating an `Ip` with that variable: `create(:ip, location_id: location.id, address: ip)`

You could do this like:
- Create an Ip: `let(:ip) { create(:ip) }`
- creating three sessions referencing that variable Session.create!(siteIP: ip.address)`

Less code, more logical and much easier to follow

Reply:
I need to set an IP with an address as I need to search for that IP